ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/Client.java

```diff
@@ -71,9 +71,32 @@
    */
   BatchedQueryResult executeQuery(String sql, Map<String, Object> properties);
 
-  CompletableFuture<Void> insertInto(String streamName, Map<String, Object> row);
+  /**
+   * Inserts a row into a ksqlDB stream.
+   *
+   * <p>The {@code CompletableFuture} will be failed if a non-200 response is received from the
+   * server, or if the server encounters an error while processing the insertion.
+   *
+   * @param streamName name of the target stream
+   * @param row the row to insert. Keys are column names and values are column values.
+   * @return a future that completes once the server response is received
+   */
+  CompletableFuture<Void> insertInto(String streamName, KsqlObject row);
+
+  /**
+   * Inserts the specified row(s) into a ksqlDB stream.
+   *
+   * <p>The {@code CompletableFuture} will be failed if a non-200 response is received from the
+   * server, or if the server encounters an error while processing the insertion(s).
+   *
+   * @param streamName name of the target stream
+   * @param rows the rows to insert. For each row, the keys are column names and values are
+   *        column values.
+   * @return a future that completes once the server response is received
+   */
+  CompletableFuture<Void> insertInto(String streamName, List<KsqlObject> rows);
```

Comment:
I don't think it's useful to have an insertInto method that takes a List of Rows. When inserting rows, the actual inserts can occur our of order, and it's possible that some inserts could succeed and some could fail. Returning a single CompletableFuture means that we couldn't tell the user about individual success or failure of inserts. So on failure, it might be that some of the inserts actually succeeded, leaving the system in an inconsistent state and the user not being able to do anything useful to correct it.

At some point I hope we will support transactions on the client. At that point I think it would make sense to support batch inserts like this, but not until then. With transactions we can guarantee that all the inserts succeed or none.

Reply:
This is the reason I didn't include a batch insert method like this on the original prototype.

Reply:
Hmm OK. My reasoning for adding the batch insert was that it seems inefficient to require the users to open a separate HTTP connection for each insert, but I guess once we implement the streaming insert we can encourage them to use that instead. I'll remove the batch insert method in light of your concern that the non-transactional nature will be confusing.

Reply:
> a separate HTTP connection for each insert

The client pools connections so it won't use a new one for each insert.